modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/Range.scala

```diff
@@ -0,0 +1,70 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.{Validated, ValidatedNel}
+import cats.syntax.all._
+
+import scala.util.control.NoStackTrace
+
+import derevo.cats.show
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+
+case class InvalidOrdinal(ordinal: String) extends NoStackTrace {
+  override def getMessage = s"Invalid ordinal: $ordinal"
+}
+
+case class InvalidRangeFormat(range: List[String]) extends NoStackTrace {
+  override def getMessage = s"Invalid range format: ${range.mkString("-")}"
+}
+
+case class ReversedRange(start: NonNegLong, end: NonNegLong) extends NoStackTrace {
+  override def getMessage =
+    s"Invalid range: '$start-$end'. The start value must not be greater than the end value."
+}
+
+@derive(show)
+case class Range(start: NonNegLong, end: NonNegLong)
+
+object Range {
+
+  type ValidationResult[A] = ValidatedNel[Throwable, A]
+
+  private val delimiter = "-"
+
+  def decode(input: String): ValidationResult[Range] = {
+    val values = input.split(delimiter).map(_.trim).toList
+
+    parseAndValidate(values)
+  }
+
+  private def parseAndValidate(inputs: List[String]): ValidationResult[Range] =
+    parseRange(inputs).andThen { r =>
+      Validated.condNel(
+        r.start <= r.end,
+        r,
+        ReversedRange(r.start, r.end)
+      )
```

Comment:
didn't we agree on the meeting that if you decouple things, the reversed range is a totally valid range in here, where you only parse the range?

Reply:
Moved the valid `Range` checks to `ranges.decode`.